File: testdata/fixtures/ai/python/detect-gemini.py
# AI usage detection: Gemini
def vuln():
    # ruleid: detect-gemini
    import google.generativeai

    def inner():
        # ruleid: detect-gemini
        import google.generativeai

    try:
        # ruleid: detect-gemini
        import google.generativeai
    except ImportError:
        pass


def safe():
    # ok: detect-gemini
    import os
    # ok: detect-gemini
    import numpy
    # ok: detect-gemini
    data = "safe string"

File: testdata/fixtures/ai/go/detect-gemini.go
// AI usage detection: Gemini via generative-ai-go
package main
// ruleid: detect-gemini
import "github.com/google/generative-ai-go"
import "fmt"
// ruleid: detect-gemini
func newGemini() { genai.NewClient(nil) }
// ruleid: detect-gemini
func otherGemini() { genai.NewClient("key") }
// ok: detect-gemini
import "strings"
// ok: detect-gemini
func safe() { fmt.Println("hi") }
// ok: detect-gemini
func safe2() { return }
